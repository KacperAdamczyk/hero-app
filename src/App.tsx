import React, { FC } from 'react';
import styled from '@emotion/styled';
import { ThemeProvider } from '@material-ui/core';
import {
  BrowserRouter as Router,
  Routes,
  Route,
  Redirect,
} from 'react-router-dom';
import { ApolloProvider } from '@apollo/react-hooks';

import { colors, theme, media } from 'styling';
import { client } from 'api';
import { HeroesList, AddHero, HeroDetails } from 'pages';

const Container = styled.div`
  background-color: ${colors.background};
  box-sizing: border-box;
  min-height: 100vh;
  padding: 15px 70px;

  ${media.small} {
    padding: 15px 10px;
  }

  ${media.medium} {
    padding: 15px 35px;
  }
`;

export const App: FC = () => (
  <ApolloProvider client={client}>
    <ThemeProvider theme={theme}>
      <Router>
        <Container>
          <Routes>
            <Redirect from="/" to="heroes" />
            <Route path="heroes/*" element={<HeroesList />}>
              <Route path="add" element={<AddHero />} />
              <Route path=":id" element={<HeroDetails />} />
            </Route>
          </Routes>
        </Container>
      </Router>
    </ThemeProvider>
  </ApolloProvider>
);
