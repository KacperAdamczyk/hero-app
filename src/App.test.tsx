import React from 'react';
import { render } from '@testing-library/react';
import { App } from './App';

jest.mock('react-modal');

test('renders layout', () => {
  const { container } = render(<App />);

  expect(container.firstChild).toMatchSnapshot();
});
