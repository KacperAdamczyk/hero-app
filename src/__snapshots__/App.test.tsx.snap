// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders layout 1`] = `
<div
  class="css-7bpeju"
>
  <div
    class="css-vzmcve-HeroesList"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained css-1lu5or6-Button MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        Add a hero
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="css-cssveg"
  >
    <div
      class="css-4c3f7w"
    />
    <div
      class="css-1jw2zp1-Loader"
      data-testid="loader"
    >
      <div
        class="css-0"
      >
        <div
          class="css-19ve4cl"
        />
        <div
          class="css-4wm62s"
        />
        <div
          class="css-19ve4cl"
        />
      </div>
    </div>
  </div>
</div>
`;
